app/src/test/java/ch/hikemate/app/model/facilities/FacilitiesViewModelTest.kt

```diff
@@ -0,0 +1,183 @@
+import org.osmdroid.util.GeoPoint
+
+class FacilitiesViewModelTest {
+
+  private lateinit var mockFacilitiesRepository: FacilitiesRepository
+  private lateinit var testDispatcher: TestDispatcher
+
+  private lateinit var facilitiesViewModel: FacilitiesViewModel // SUT
+
+  private val testBounds = Bounds(46.5, 6.6, 46.51, 6.62)
+  private val testFacilities = listOf(Facility(FacilityType.TOILETS, GeoPoint(46.51, 6.61)))
+
+  @OptIn(ExperimentalCoroutinesApi::class)
+  @Before
+  fun setUp() {
+    testDispatcher = StandardTestDispatcher()
+    Dispatchers.setMain(testDispatcher)
+
+    mockFacilitiesRepository = mock()
+    facilitiesViewModel = FacilitiesViewModel(mockFacilitiesRepository, testDispatcher)
+  }
+
+  @OptIn(ExperimentalCoroutinesApi::class)
+  @After
+  fun tearDown() {
+    Dispatchers.resetMain()
+  }
+
+  @Test
+  fun testGetFacilities_onSuccess() {
+    val facilities = listOf(Facility(FacilityType.TOILETS, GeoPoint(46.51, 6.61)))
+
+    `when`(mockFacilitiesRepository.getFacilities(any(), any(), any())).then {
+      val onSuccess = it.getArgument<(List<Facility>) -> Unit>(1)
+      onSuccess(facilities)
+    }
+
+    facilitiesViewModel.getFacilities(
+        bounds = testBounds,
+        onSuccess = { result -> assertEquals(facilities, result) },
+        onFailure = { fail("Should not be called") })
+  }
+
+  @Test
+  fun testGetFacilities_onFailure() {
+    val exception = Exception("Test exception")
+
+    `when`(mockFacilitiesRepository.getFacilities(any(), any(), any())).then {
+      val onFailure = it.getArgument<(Exception) -> Unit>(2)
+      onFailure(exception)
+    }
+
+    facilitiesViewModel.getFacilities(
+        bounds = testBounds,
+        onSuccess = { fail("Should not be called") },
+        onFailure = { ex -> assertEquals(exception, ex) })
+  }
+
+  @Test
+  fun testGetFacilities_usesCache_sameBounds() = runTest {
+    val facilities = listOf(Facility(FacilityType.TOILETS, GeoPoint(46.51, 6.61)))
+
+    var onSuccessCallCount = 0 // Used to make sure the first onSuccess callback is only called once
+
+    `when`(mockFacilitiesRepository.getFacilities(any(), any(), any())).then {
+      val onSuccess = it.getArgument<(List<Facility>) -> Unit>(1)
+      onSuccess(facilities)
+    }
+
+    facilitiesViewModel.getFacilities(
+        bounds = testBounds,
+        onSuccess = { result ->
+          assertEquals(facilities, result)
+          onSuccessCallCount++
+        },
+        onFailure = { fail("Should not be called") })
+
+    testDispatcher.scheduler.advanceUntilIdle()
+
+    // Second call to getFacilities should use the cache
+    facilitiesViewModel.getFacilities(
+        bounds = testBounds,
+        onSuccess = { result -> assertEquals(facilities, result) },
+        onFailure = { fail("Should not be called") })
+
+    testDispatcher.scheduler.advanceUntilIdle()
+
+    verify(mockFacilitiesRepository, times(1)).getFacilities(any(), any(), any())
+    assertEquals(1, onSuccessCallCount)
```

Comment:
Warning: You should have 2 success calls (and therefore add an incrementation in the second call

Reply:
You're absolutely right. I've added a separate variable for each call, to ensure each one is only called once.